db.go

```diff
@@ -330,20 +337,38 @@ func (x *dummyUserStore) GetIdentities(getIdentitiesFlag GetIdentitiesFlag) ([]A
 		if (getIdentitiesFlag&GetIdentitiesFlagDeleted == 0) && v.archived {
 			continue
 		}
-		list = append(list, AuthUser{v.userId, v.email, v.username, v.firstname, v.lastname, v.mobilenumber, v.telephonenumber, v.remarks, v.created, v.createdby, v.modified, v.modifiedby, v.authUserType, v.archived, v.passwordModifiedDate})
+		list = append(list, AuthUser{v.userId, v.email, v.username, v.firstname, v.lastname, v.mobilenumber, v.telephonenumber, v.remarks, v.created, v.createdby, v.modified, v.modifiedby, v.authUserType, v.archived, v.passwordModifiedDate, v.accountLocked})
 	}
 	return list, nil
 }
 
+func (x *dummyUserStore) LockAccount(userId UserId) error {
+	x.usersLock.RLock()
+	defer x.usersLock.RUnlock()
```

Comment:
This is a ReadWrite lock, but you're modifying the value here, so you need to take the full write-able version of the lock.
Read about the semantics of a ReadWrite lock.